pkg/util/validation/validate.go

```diff
@@ -238,3 +241,27 @@ func formatLabelSet(ls []client.LabelAdapter) string {
 
 	return fmt.Sprintf("%s{%s}", metricName, strings.Join(labelStrings, ", "))
 }
+
+func DeletePerUserValidationMetrics(userID string) {
```

Comment:
[nit, non-blocking] Given we're moving away from the global logger, what do you think taking the logger in input?